pennylane/devices/device_api.py

```diff
@@ -323,12 +323,133 @@ def execute_fn(tapes):
             Only then is the classical postprocessing called on the result object.
 
         """
-        if self.supports_derivatives(execution_config) and execution_config.gradient_method in {
-            "best",
-            None,
-        }:
-            return TransformProgram(), replace(execution_config, gradient_method="device")
-        return TransformProgram(), execution_config
+        execution_config = self.setup_execution_config(execution_config)
+        transform_program = self.preprocess_transforms(execution_config)
+        return transform_program, execution_config
+
+    def setup_execution_config(
+        self, config: ExecutionConfig = DefaultExecutionConfig
```

Comment:
This might also be a good time time to make the change to `config: Optional[ExecutionConfig] = None`.  Something we need to do soon, since a mutable default value is a bad idea, but we've just been ignoring it for the moment.

Reply:
Didn't we have discussions about making the execution config immutable?

Reply:
Yes.  We decided we needed to do so, and want to do so before the next release, but just didn't immediately prioritize it do the number of things going on.

Reply:
I'll change it to None and initialize the default config within the method for now.

Reply:
Done

Reply:
Thanks :)